### PR TITLE
Better publication docs

### PR DESCRIPTION
**Pull request by dwil**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -63,14 +63,69 @@ Publishing
     git tag 0.5.1
     git push origin 0.5.1
 
-4. Register the package::
+``.pypirc`` file
+-----------------
+
+To ensure publication and testing of publications are submitted to the correct
```
Comment (bmacphee) [inline README.rst]: hard to parse
does this mean
`To ensure publication and to test that publications are submitted to...`?

Or are we talking about submitting publications and testing of those publications?  I am confused.

File: README.rst [hunk 2]
```diff
@@ -63,14 +63,69 @@ Publishing
...
+    twine upload dist/* -r testpypi
+
+5. Test that you can install the package from testpypi::
+
+    pip install -i https://testpypi.python.org/pypi analyzere
+
+Publishing
+-----------
+
+1. Ensure you have tagged the master repository with the according to the
```
Comment (bmacphee) [inline README.rst]: `Ensure you have tagged the master repository with the according to the`

=> 

`Ensure you have tagged the master repository according to the`

File: README.rst (hunk 1 above)
Comment (dwil) [inline README.rst]: What I'm attempting to say is the `.pypirc` is used to ensure that you can submit your package to the correct server where the cases are:
- I want to test publishing my package (i.e. I send it to a "dummy" server and then I can test that I can `pip install` etc). In this case I will use the `testpypi` server.
- I want to actually publish my package. In which case I will use the `pypi` server.

The unfortunate part is doing commands like: 
`python setup.py register -r testpypi`
`twine upload dist/* -r pypi`

with the `-r` argument only seem to work if you have a `.pypirc` file created.

Comment (DanielArndt) [inline README.rst]: I would just skip the explanation and say "Configure your `.pypirc` file for your production and test server accounts. This file should look as follows:", by the time you follow through the rest of the guide, it should be apparent what this is for. That's my two cents.

File: README.rst [hunk 3]
```diff
@@ -63,14 +63,69 @@ Publishing
...
+    username = <username>
+    password = <password>
+
+
+Note that ``testpypi`` and ``pypi`` require separate registration.
+
+Testing Publication
+-----------------
+
+1. Ensure you have tagged the master repository with the according to the
```
Comment (DanielArndt) [inline README.rst]: ~~with the~~
